pkg/awsutils/awsutils.go

```diff
@@ -845,33 +861,117 @@ func (cache *EC2InstanceMetadataCache) createENI(useCustomCfg bool, sg []*string
 			SecondaryPrivateIpAddressCount: aws.Int64(int64(needIPs)),
 		}
 	}
-	if useCustomCfg {
-		log.Info("Using a custom network config for the new ENI")
-		if len(sg) != 0 {
-			input.Groups = sg
+
+	var err error
+	var networkInterfaceID string
+	if cache.useCustomNetworking {
+		input = createENIUsingCustomCfg(sg, eniCfgSubnet, input)
+		log.Infof("Creating ENI with security groups: %v in subnet: %s", aws.StringValueSlice(input.Groups), aws.StringValue(input.SubnetId))
+
+		networkInterfaceID, err = cache.tryCreateNetworkInterface(input)
+		if err == nil {
+			return networkInterfaceID, nil
+		}
+	} else {
+		if cache.useAdvancedENIConfig {
+			subnetResult, vpcErr := cache.getVpcSubnets()
+			if vpcErr != nil {
+				return "", errors.Wrap(vpcErr, "failed to get subnets in vpc")
+			}
+
+			for _, subnet := range subnetResult {
```

Comment:
This can be simplified, something like -

if (enhanced_subnet_selection) {
    1. get a subnet
    2. Set input.SubnetID
}

Then call ->
networkInterfaceID, err = cache.tryCreateNetworkInterface(input)